compiler/damlc/tests/daml-test-files/Daml3ScriptTrySubmit.daml

```diff
@@ -0,0 +1,197 @@
+  alice <- allocateParty "alice"
+  let t3Key = (alice, [])
+  res <- alice `trySubmit` createAndExerciseCmd (FetchByKeyT3 with fetcher = alice, fetchKey = t3Key) DoFetch
+  case res of
+    Left (FetchEmptyContractKeyMaintainers (fromAnyContractKey @T3 -> Some shouldBeT3Key)) -> shouldBeT3Key === t3Key
+    Left e -> error $ "fetchEmptyContractKeyMaintainers incorrect error: " <> show e
+    Right _ -> error "fetchEmptyContractKeyMaintainers incorrectly succeeded"
+
+wronglyTypedContract : Script ()
+wronglyTypedContract = script do
+  alice <- allocateParty "alice"
+  t1Cid <- alice `submit` createCmd T1 with t1_p = alice
+  let t2Cid = coerceContractId @T1 @T2 t1Cid
+  res <- alice `trySubmit` exerciseCmd t2Cid Archive
+  case res of
+    Left (WronglyTypedContract (fromAnyContractId @T1 -> Some shouldBeT1Cid) expectedTypeRep actualTypeRep) -> do
+      shouldBeT1Cid === t1Cid
+      expectedTypeRep === templateTypeRep @T2
+      actualTypeRep === templateTypeRep @T1
+    Left e -> error $ "wronglyTypedContract incorrect error: " <> show e
+    Right _ -> error "wronglyTypedContract incorrectly succeeded"
+
+-- ContractDoesNotImplementInterface: Can't do this from daml-script with IDELedger
+-- ContractDoesNotImplementRequiringInterface: Can't do this from daml-script with IDELedger
+-- NonComparableValues: Can't do this from daml-script with IDELedger
+-- ContractIdInContractKey: Can't do this from daml-script with IDELedger
+-- ContractIdComparability: Can't do this from daml-script with IDELedger
+
+devError : Script ()
+devError = script do
+  alice <- allocateParty "alice"
+  cid <- alice `submit` createCmd T1 with t1_p = alice
+  res <- alice `trySubmit` exerciseCmd (toInterfaceContractId @ExerciseGuardedInterface cid) ExerciseGuarded with shouldArchive = False
+  case res of
+    Left (DevError ChoiceGuardFailed msg) -> assert $ "Choice guard failed for" `isInfixOf` msg
+    Left e -> error $ "devError incorrect error: " <> show e
+    Right _ -> error "devError incorrectly succeeded"
+
+-- UnknownError: Shouldn't be testable, we should know all the possible errors.
+-- Can reproduce right now with a timeout, or package vetting test.
+
+-- TruncatedError: Even with below test, creating a key with 100,000 parties, it doesn't hit the limit somehow.
+-- I don't know how to hit the payload limit
+
+-- -- Using contractKeyNotFound with an enormous key
+-- truncatedError : Script ()
+-- truncatedError = script do
+--   alice <- allocateParty "alice"
+--   let t4Key = (alice, replicate 100000 alice)
+--   res <- trySubmit [alice] [alice] $ exerciseByKeyCmd @T4 t4Key Archive
+--   case res of
+--     Left (TruncatedError "ContractKeyNotFound" msg) -> error msg
+--     Left (ContractKeyNotFound (fromAnyContractKey @T4 -> Some shouldBeT4Key)) | shouldBeT4Key == t4Key -> error "It sent the whole thing??"
+--     Left e -> error $ "contractKeyNotFound incorrect error: " <> show e
+--     Right _ -> error "contractKeyNotFound incorrectly succeeded"
```

Comment:
Depending on if we want to keep Truncated error, would love ideas to resolve this

Reply:
I've been a fool - we're not truncating on the IDELedger, this could only fail in canton tests.

Reply:
TODO: Update this so it throws a specific error on truncation, and assert that error in the script runner test